plugin/plugin.go

```diff
@@ -141,7 +141,7 @@ func (p *Plugin) Init(ctx context.Context, spec []byte, options NewClientOptions
 	defer p.mu.Unlock()
 	var err error
 
-	if p.schemaValidator != nil {
+	if len(spec) > 0 && p.schemaValidator != nil {
```

Comment:
Shouldn't we still validate the spec even if it is nil?

Reply:
I think we may need to move the validation somewhere else, or pass in an option to skip validation, if we need to init a plugin without running validation

Reply:
I've moved the check to `NoConnection` option setting, as it's the sole case for our docs gen